### ISSUE TITLE
Processing inference results

### ISSUE THREAD

**Issue by IgorGoldberg** [ultralytics/ultralytics#7348]
### Search before asking

- [ ] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Let's say I did inference and stored it to a variable like this:
results = model.predict(source,
              show_boxes = True,
              save = False,#!!!
              show_labels = True,
              show_conf = True,
              conf = 0.5,
              save_txt = False,#!!!
              save_crop = False,
              line_width = 2,
              iou = 0.5,
              )
Now I want to filter out some instances.
For example, my condition is 
if the bounding box area > X and class =="b":
 then I want to delete this instance and all its relative information. 
 
NOTE :
**I DO KNOW how to iterate through instances in results** and check things like class name and calculate the area of the bounding box etc

what I do NOT know is how to :
1)how to delete from results all the information regarding some instance that answers my criteria as in the example above
2)after I updated the results by deleting some instances is there a QUICK way to store updated images and text labels based on updated results - the results where I had deleted the unneeded instances?

Thank you very much

**Comment by Burhan-Q** [comment]
@IgorGoldberg personally I wouldn't modify the `results` object directly but instead make copies or views of the data to use for filtering. That said, here's how you could modify the `results` data.

```py
import torch
from ultralytics import YOLO

model = YOLO('yolov8n.pt')
results = model.predict('bus.jpg')

new_results = results.copy()
for ri,result in enumerate(results):
    b = result.boxes.data.view(result.boxes.data.shape) # copy
    
    filtering = ... # run your filtering, output indices to keep
    idx = [[i] for i in filtering] # NOTE each index needs to be in this form
    # Example (keeping only indices [1, 3, 4, 5])
    # idx = [[1], [3], [4], [5]]
    
    new_boxes = b.take_along_dim(torch.tensor(idx, device=result.boxes.data.device), 0)
    result.update(boxes=new_boxes)
    new_results[ri] = result
```
    
Note that I didn't put any filtering functions as I'm sure you have something in mind for what needs to be done. The only requirement for you to consider in noted in the comments about the index formatting to use with the `take_along_dim` method.

FYI, if you want to filter _all_ instances of a class or list of classes, you can use the argument `classes=[3]` to prevent any detections for that class from getting added into `results`. The Docs section on [Inference Arguments](https://docs.ultralytics.com/modes/predict/#inference-sources) is a good reference point.

Finally, this is certainly a good topic to discuss with members of our <a href="https://ultralytics.com/discord"><img alt="Discord" src="https://img.shields.io/discord/1089800235347353640?logo=discord&logoColor=white&label=Discord&color=blue"></a> server. I'm moderator there and there's an active community of users discussing how to accomplish things like this. Hopefully it helps and look forward to seeing you on the Discord!
